lib/components/map/gtfs-rt-vehicle-overlay.js

```diff
@@ -0,0 +1,218 @@
+    liveFeedUrl: PropTypes.string,
+    /** URL to GTFS-like route list in JSON format. */
+    routeDefinitionUrl: PropTypes.string
+  }
+
+  constructor (props) {
+    super(props)
+    this.state = {
+      routes: [],
+      vehicleLocations: [],
+      // Set to undefined to fall back on the default symbols,
+      // unless a route definition is provided.
+      vehicleSymbols: undefined,
+      visible: props.visible
+    }
+  }
+
+  componentDidMount () {
+    const { liveFeedUrl, name, registerOverlay, routeDefinitionUrl } = this.props
+    registerOverlay(this)
+    if (!liveFeedUrl) {
+      console.warn(`liveFeedUrl prop is missing for overlay '${name}'.`)
+    }
+    if (!routeDefinitionUrl) {
+      console.warn(`routeDefinitionUrl prop is missing for overlay '${name}'.`)
+    } else {
+      this._fetchRoutes()
+    }
+  }
+
+  componentWillUnmount () {
+    this._stopRefreshing()
+  }
+
+  onOverlayAdded = () => {
+    this.setState({ visible: true })
+    this._startRefreshing()
+  }
+
+  onOverlayRemoved = () => {
+    this.setState({ visible: false })
+    this._stopRefreshing()
+  }
+
+  createLeafletElement () {}
+
+  updateLeafletElement () {}
+
+  /**
+   * Fetches GTFS-rt vehicle positions (protocol buffer) and converts to
+   * OTP-UI transitVehicleType before saving to component state.
+   */
+  _fetchVehiclePositions = async () => {
+    if (this.props.liveFeedUrl) {
+      try {
+        const response = await fetch(this.props.liveFeedUrl)
+        if (response.status >= 400) {
+          const error = new Error('Received error from server')
+          error.response = response
+          throw error
+        }
+        const buffer = await response.arrayBuffer()
+        const view = new Uint8Array(buffer)
+        const feed = transitRealtime.FeedMessage.decode(view)
+
+        const vehicleLocations = feed.entity.map(vehicle => gtfsRtToTransitVehicle(vehicle, this.state.routes))
+        this.setState({ vehicleLocations })
+      } catch (err) {
+        console.log(err)
+      }
+    }
+  }
+
+  /**
+   * Fetches GTFS-like route definitions (JSON).
+   * (called once when component is mounted)
+   */
+  _fetchRoutes = async () => {
+    if (this.props.routeDefinitionUrl) {
+      try {
+        const response = await fetch(this.props.routeDefinitionUrl)
+        if (response.status >= 400) {
+          const error = new Error('Received error from server')
+          error.response = response
+          throw error
+        }
+        const routes = await response.json()
+
+        // Generate the symbols for the overlay at this time
+        // so it renders the route colors for each vehicle.
+        const vehicleSymbols = [
+          {
+            minZoom: 0,
+            symbol: makeVehicleShape(routes)
+          }
+        ]
+
+        this.setState({ routes, vehicleSymbols })
+      } catch (err) {
+        console.log(err)
+      }
+    }
+  }
+
+  _startRefreshing () {
+    // ititial vehicle retrieval
```

Comment:
```suggestion
// initial vehicle retrieval
```
small typo!

Reply:
Fixed in https://github.com/opentripplanner/otp-react-redux/pull/401/commits/4348cdb2c20d2658d9b4f113dd15e18d3b110939